packages/form-field/src/FormField.spec.tsx

```diff
@@ -294,123 +295,128 @@ describe('packages/form-field', () => {
+  describe(`when state is not ${FormFieldState.None}`, () => {
+    test('form field feedback has an id', () => {
+      const { getByTestId } = render(
         <FormField
           label="Label"
           successMessage="Success"
+          state={FormFieldState.Valid}
           data-testid="form-field"
         >
           <FormFieldInputContainer>
             <div data-testid="input" />
           </FormFieldInputContainer>
         </FormField>,
       );
-      const successIcon = queryByTitle('Valid');
-      const successText = queryByText('Success');
-      expect(successIcon).toBeNull();
-      expect(successText).toBeNull();
+      const feedback = getByTestId(LGIDS_FORM_FIELD.feedback);
+      expect(feedback).toHaveAttribute('id');
     });
 
-    test(`Success icon and message appear when state is ${FormFieldState.Valid}`, () => {
-      const { queryByText, queryByTitle } = render(
+    test('input is described by form field feedback', () => {
+      const { getByTestId } = render(
         <FormField
           label="Label"
-          successMessage="Success"
-          state={FormFieldState.Valid}
+          errorMessage="This is an error message"
+          state={FormFieldState.Error}
           data-testid="form-field"
         >
           <FormFieldInputContainer>
             <div data-testid="input" />
           </FormFieldInputContainer>
         </FormField>,
       );
-      const successIcon = queryByTitle('Valid');
-      const successText = queryByText('Success');
-      expect(successIcon).toBeInTheDocument();
-      expect(successText).toBeInTheDocument();
+      const input = getByTestId('input');
+      const feedback = getByTestId(LGIDS_FORM_FIELD.feedback);
+      expect(input).toHaveAttribute('aria-describedby', feedback?.id);
     });
-  });
 
-  describe(`${FormFieldState.Error} state`, () => {
-    test('Error icon and message are not shown by default', () => {
-      const { queryByText, queryByTitle } = render(
+    test('input is described by both description & feedback', () => {
+      const { getByTestId, queryByText } = render(
         <FormField
           label="Label"
-          errorMessage="This is an error message"
+          description="Description"
+          successMessage="Success"
+          state={FormFieldState.Valid}
           data-testid="form-field"
         >
           <FormFieldInputContainer>
             <div data-testid="input" />
           </FormFieldInputContainer>
         </FormField>,
       );
-      const errorIcon = queryByTitle('Error');
-      const errorText = queryByText('This is an error message');
-      expect(errorIcon).toBeNull();
-      expect(errorText).toBeNull();
+      const input = getByTestId('input');
+      const description = queryByText('Description');
+      const feedback = getByTestId(LGIDS_FORM_FIELD.feedback);
+      expect(input).toHaveAttribute(
+        'aria-describedby',
+        description?.id + ' ' + feedback?.id,
```

Comment:
If we can test this in a way where the order of ids doesn't matter, that'd be better (this is technically testing a subtle implementation detail)

Reply:
https://github.com/mongodb/leafygreen-ui/pull/2318/commits/3b444b2d490b74792fc8d9972616b1d6681d4db1#diff-96f34579a072b062c22857b1e854bfefef73318f69c1d41de310cfb2d77bdb45L353-R357